src/components/clusterprovider/minikube/minikubeclusterprovider.ts

```diff
@@ -0,0 +1,224 @@
+    const sr = await context.shell.exec(cmd);
+
+    const createCluster = await fromShellExitCodeOnly(sr);
+    return {
+        actionDescription: 'creating cluster',
+        result: createCluster
+    };
+}
+
+async function createCluster(previousData: any, context: Context): Promise<string> {
+    const createResult = await runMinikubeCommand(context, 'minikube help');
+
+    context.createExec = context.shell.exec('minikube start');
+
+    const title = createResult.result.succeeded ? 'Cluster creation has started' : `Error ${createResult.actionDescription}`;
+    const additionalDiagnostic = diagnoseCreationError(createResult.result);
+    const successCliErrorInfo = diagnoseCreationSuccess(createResult.result);
+    const message = succeeded(createResult.result) ?
+        `<div id='content'>
+         ${formStyles()}
+         ${styles()}
+         <!-- ${waitScript('Checking on minikube cluster')} -->
+         <form id='form' action='create?step=wait' method='post' onsubmit='return promptWait();'>
+         ${propagationFields(previousData)}
+         <p class='success'>Minikube is creating the cluster, but this may take some time. You can now close this window,
+         or wait for creation to complete so that we can add the new cluster to your Kubernetes configuration.</p>
+         <p><button type='submit' class='link-button'>Wait and add the new cluster &gt;</button></p>
+         </form>
+         ${successCliErrorInfo}
+         </div>` :
+        `<p class='error'>An error occurred while creating the cluster. Is 'minikube' installed and in your PATH?</p>
+         ${additionalDiagnostic}
+         <p><b>Details</b></p>
+         <p>${createResult.result.error[0]}</p>`;
+    return `<!-- Complete -->
+            <h1 id='h'>${title}</h1>
+            ${styles()}
+            ${waitScript('Waiting for cluster - this will take several minutes')}
+            ${message}`;
+
+}
+
+async function waitForClusterAndReportConfigResult(context: Context, previousData: any): Promise<string> {
+    const waitResult = await runMinikubeCommand(context, 'minikube status');
+    if (!waitResult.result.succeeded) {
+        const failed = waitResult.result as Failed;
+        return `<h1>Waiting for minikube cluster</h1>
+        <p>Current Status</p>
+        <pre><code>${failed.error[0]}</code></pre>
+        <form id='form' action='create?step=wait' method='post'>
+        ${propagationFields(previousData)}
+        </form>
+        <script>
+        window.setTimeout(function() {
+            var f = document.getElementById('form');
+            f.submit();
+        }, 1000)
+        </script>
+        `;
+    }
+    await refreshExplorer();
+
+    return renderConfigurationResult();
+}
+
+function renderConfigurationResult(): string {
+    const title = 'Cluster added';
+    const getCliOutput = '';
+    const getCredsOutput = '';
+    return `<!-- Complete -->
```

Comment:
Why bother with `getCliOutput` and `getCredsOutput`?  Those were relevant to ACS/AKS where we made two Azure CLI calls (to download the kubectl CLI and to fetch the cluster credentials into kubeconfig).  There's nothing contractual about that implementation detail; we don't need to stub it out here.

Reply:
removed (cargo cult)